core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -831,6 +835,14 @@ private[spark] class AppStatusListener(
         }
       }
     }
+    // Flush updates if necessary. Executor heartbeat is an event that happens periodically. Flush
+    // here to ensure the staleness of Spark UI doesn't last more that the executor heartbeat
+    // interval.
+    if (now - lastFlushTimeNs > liveUpdatePeriodNs) {
```

Comment:
I'm also worried about the case when `flush()` takes a few milliseconds to finish, and you end up with always dealing with updating all live entities for each ExecutorMetricsUpdate event.
Is it possible to introduce a new config that specifies the live update period for ExecutorMetricsUpdate only? The default value can be the same as `liveUpdatePeriodNs`, while user can change it to a bigger value when the `flush()` function become a issue in processing the event.